src/apps/vpn/controller.cpp

```diff
@@ -234,32 +234,39 @@ bool Controller::activate(const ServerData& serverData,
     request->get(AppConstants::apiUrl(AppConstants::Account));
 
     connect(request, &NetworkRequest::requestFailed, this,
-            [this](QNetworkReply::NetworkError error, const QByteArray&) {
+            [this, serverSelectionPolicy](QNetworkReply::NetworkError error,
+                                          const QByteArray&) {
               logger.error() << "Account request failed" << error;
               REPORTNETWORKERROR(error, ErrorHandler::DoNotPropagateError,
                                  "PreActivationSubscriptionCheck");
 
               // Check if the error propagation has changed the Mozilla VPN
               // state. Continue only if the user is still authenticated and
               // subscribed.
-              if (MozillaVPN::instance()->state() == MozillaVPN::StateMain) {
-                setState(StateConnecting);
-              } else {
-                setState(StateOff);
+              if (MozillaVPN::instance()->state() != MozillaVPN::StateMain) {
+                return;
               }
+
+              setState(StateConnecting);
+
+              clearRetryCounter();
```

Comment:
Can you explain about why this and the next line is needed?

Reply:
This line should be moved elsewhere, but I did not want to apply big changes in this PR and just fix the crash.
This line resets the Controller timer (what we show in the UI). 
We don't want to move it in activateInternal() because that means it will be seen during silent-switch too. But still, it's horrible to have it around.